test/watch/test/global-setup-rerun.test.ts

```diff
@@ -0,0 +1,27 @@
+import { expect, test } from 'vitest'
+import { editFile, runVitest } from '../../test-utils'
+
+const testFile = 'fixtures/math.test.ts'
+
+test('global setup calls hooks correctly', async () => {
+  process.env.TEST_GLOBAL_SETUP = 'true'
+  const { vitest, ctx } = await runVitest({
+    root: 'fixtures',
+    watch: true,
+    include: ['math.test.ts'],
+  })
+
+  await vitest.waitForStdout('Waiting for file changes')
+
+  const calls = (globalThis as any).__CALLS as string[]
+  expect(calls).toEqual(['start'])
+
+  editFile(testFile, testFileContent => `${testFileContent}\n\n`)
```

Comment:
I think we could use `'r'` keypress here instead. Editing files is heavy on CI.

```ts
  vitest.write('r')

  await vitest.waitForStdout('RERUN')
```

Reply:
But we are testing that _watcher_ triggered a callback

Reply:
There should be two tests then, not one

Reply:
If we are testing file editing here, then yes. If it's just about testing whether rerun calls this callback, then it's not important.

Reply:
We are testing everything that affects this, I don't think about implementation details when testing it. When user does something, something else should happen